src/main/java/org/intellij/sdk/codesync/Utils.java

```diff
@@ -45,6 +45,11 @@ public static Boolean shouldSkipEvent(String repoPath) {
         }
     }
 
+    public static boolean isIndividualFileOpen(String contentRootPath) {
+        File file = new File(contentRootPath);
```

Comment:
Awesome 👍🏼 I think this is better than my suggestion. 
Make sure to test out all the flows though.

Reply:
Also, I believe we don't need to put check in detectBranchChange in populate anymore because it will not be reachable as single file repo will not be already synced.